tests/tests_geomstats/test_pullbackdiffeo_metric.py

```diff
@@ -0,0 +1,219 @@
+"""Unit tests for the pull-back diffeo metrics."""
+
+import random
+
+import geomstats.backend as gs
+import geomstats.tests
+from geomstats.geometry.hypersphere import Hypersphere
+from geomstats.geometry.pullback_metric import PullbackDiffeoMetric
+from geomstats.geometry.special_orthogonal import SpecialOrthogonal
+from tests.conftest import Parametrizer
+from tests.data.hypersphere_data import HypersphereMetricTestData
+from tests.data.pullback_diffeo_metric_data import PullbackDiffeoCircleMetricTestData
+from tests.geometry_test_cases import PullbackDiffeoMetricTestCase
+from tests.tests_geomstats.test_hypersphere import AbstractHypersphereMetric
+
+
+class CircleAsSpecialOrthogonalMetric(PullbackDiffeoMetric):
+    def __init__(self, *args, **kwargs):
```

Comment:
Actually, I only use *args. It is done because I use the test of the hyperspheremetric that usually take dim as positional args, but in other specific i do not, because I know it is 1. However, I agree it could be more explicit. I will change to a more straightforward version !